src/mame/audio/flower.cpp

```diff
@@ -138,18 +141,18 @@ void flower_sound_device::sound_stream_update(sound_stream &stream, std::vector<
 		int ch_volume = voice->volume;
 		int ch_frequency = voice->frequency;
 
-		if (voice->enable == false)
+		if (!voice->enable)
 			continue;
 
 		mix = &m_mixer_buffer[0];
 
 		for (int i = 0; i < buffer.samples(); i++)
 		{
-			if (voice->repeat == true)
+			if (voice->repeat)
 			{
 				raw_sample = m_sample_rom[((voice->start_address >> 7) & 0x7e00) | ((voice->position >> 7) & 0x1ff)];
-				// guess: cut off after a number of repetitions
-				if ((voice->position >> 7) & 0x20000)
+				// guess: key on/off bit is lowest bit of volume bank register?
+				if ((voice->volume_bank & 0x10) != 0x10)
```

Comment:
Changes behaviour. What's the rationale?

Reply:
Volume LUT ROM 0x2000-0x2fff area and 0x3000-0x3fff area is same (0x2000-0x3fff (or only 0x2000-0x2fff/0x3000-0x3fff) is calculated wavetable data area), and Bit 0 is cleared when wavetable SFX is ended.

Reply:
That's not a `2` tho, or maybe that's me being lost in translation.